src/parse.rs

```diff
@@ -2189,6 +2199,7 @@ pub struct Parser<'a> {
     text: &'a str,
     tree: Tree<Item<'a>>,
     refdefs: HashMap<LinkLabel<'a>, LinkDef<'a>>,
+    broken_link_callback: Option<&'a Fn(&str, &str) -> Option<(String, String)>>,
```

Comment:
Would it be possible to change the broken_link_callback to return `Option<(Cow<'a, str>, Cow<'a, str>)>` instead for performance reasons? In my use-case I don't require changing the destination, but I use the broken_link_callback more as a filter of "valid" invalid destinations.

Reply:
That sounds sensible. It's a BC break technically, but we're already backwards incompatible with regards to the `CowStr`. Maybe we can then also change the signature to take a single `&str` for which there is some reasoning in my other comment.

More input appreciated!

Reply:
Looking at the code, I don't think a `Cow` here would actually reduce allocations, I think it would need to be converted into a `String` anyway after the call. In any case, I would worry about the complexity of the type (I think you need two lifetimes, the strings separate from `'a`).